### ISSUE TITLE
Preload() efficiency problem

### ISSUE THREAD

**Issue by beef9999** [jinzhu/gorm#1411]
```
DB.Preload("Emails", "emails.deleted = 0").Find(&users)
```
```
DB.Table("users_tbl").Where("users.deleted = 0").Preload("Emails", "emails.deleted = 0").Scan(&users)
```

The first one is OK, the second reported error `(can't preload field Properties for %!s(<nil>))`

I think they are supposed to be equivalent.

----
@jbfm Thanks for your explanation, although I still want to discuss the implementation of Preload.

As I observed, Preload first select all qualified Users and then select Emails `where Emails.user_id in ("id1", "id2" ...)`

What I got here as a benchmark comparison is Python's ORM SqlAlchemy, it implements eager preloading by LEFT OUTER JOIN, and seems 2x faster than GORM. The rows scale is at approx 10k.

**Comment by jbfm** [comment]
`Scan` is part of the `Scanner` interface in the [database/sql](https://golang.org/pkg/database/sql/#Scanner) package of Go, and has nothing to do with `Find`. `Scan` is used to parse values from the SQL database into Go variables/structs.

**Comment by beef9999** [comment]
@jbfm Wouldn't they achieve the same results, in terms of filling the user structs?

The reason I choose `Scan` is that `Find` automatically filter results, if their `deleted` fields equal to 1. I want to control this behavior by myself.

**Comment by jbfm** [comment]
`Find` executes the query and populates yes, but `Scan` works on a database result, you're never executing a query by just running `Scan`. `Scan` also isn't specific to Gorm, you're simply using it wrong.

**Comment by beef9999** [comment]
@jbfm Thanks for your explanation, although I still want to discuss the implementation of Preload.

As I observed, Preload first select all qualified Users and then select Emails `where Emails.user_id in ("id1", "id2" ...)`

What I got here as a benchmark comparison is Python's ORM SqlAlchemy, it implements eager preloading by LEFT OUTER JOIN, and seems 2x faster than GORM. The rows scale is at approx 10k.
